### PR TITLE
Enable sof-docs deploy in GitHub Actions

### PR DESCRIPTION
**Pull request by xiulipan**
spilt from #321 
depend on #323 as the deploy job needs build job.

### CODE REVIEW

File: .github/actions/create-publish-folder.sh
Comment (lgirdwood) [inline .github/actions/create-publish-folder.sh:8]: missing an newline here.

File: .github/workflows/pull-request.yml
Comment (lgirdwood) [inline .github/workflows/pull-request.yml:77]: Btw, I'm assuming 20.04 since it's an LTS ?

Comment (marc-hb) [reply .github/workflows/pull-request.yml:77]: I don't think the next one is available on github yet.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:76]:
```suggestion
    needs: build
```

**Review by marc-hb — APPROVED** [review]
LGTM assuming it works

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:76]:
```suggestion
    needs: supported-reqs
```

**Comment by marc-hb** [comment]
> depend on #323 as the deploy job needs build job.

One-line change, fixed, Github Actions now passing. Please rebase #323 on top of this after it's merged. One-line conflict.

```shell
git fetch origin pull/324/head
git rebase -i HEAD~3 # two fixups
git push -f ... ghdeploy2
```

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:77]: @lgirdwood @marc-hb GitHub Actions support `ubuntu-latest` also, but now the latest is still Ubuntu 20.04 (YES, it is a LTS)
We think stick to a known version will avoid some dependency issue later.

**Comment by xiulipan** [comment]
@lgirdwood Update and fix newline issue.
@marc-hb rebased with the order that we will merge #324 first.
@deb-intel @intelkevinputnam Please help review and merge

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
